utils/src/integ-test/java/uk/gov/moj/sdt/test/utils/DBUnitUtility.java

```diff
@@ -1,635 +1,620 @@
+     *
+     * @param classToLoadFor Which class is requesting data
+     * @param reloadRefdata  the reload refdata
+     */
+    public static void loadDatabase(final Class<?> classToLoadFor, final boolean reloadRefdata) {
+        // Can have individual XML for a method, or just one for a class
+        final String fileXml = "/" + classToLoadFor.getName().replaceAll("\\.", "/") + ".xml";
+        if (DBUnitUtility.class.getResource(fileXml) != null) {
+            LOGGER.info("Loading DB with: " + fileXml);
+            DBUnitUtility.loadDatabase("SDT_OWNER", fileXml, reloadRefdata);
+            LOGGER.info("Loaded database");
+        }
+    }
+
+    /**
+     * Loads the database from a given file.
+     *
+     * @param targetSchema  Schema to load
+     * @param filePath      Path to file
+     * @param reloadRefdata the reload refdata
+     */
+    public static void loadDatabase(final String targetSchema, final String filePath, final boolean reloadRefdata) {
+        loadDatabase(targetSchema, filePath, reloadRefdata, true);
+
+    }
+
+    /**
+     * Loads the database from a given file.
+     *
+     * @param targetSchema    Schema to load
+     * @param filePath        Path to file
+     * @param reloadRefdata   the reload refdata
+     * @param cleanseDatabase clean database?
+     */
+    public static void loadDatabase(final String targetSchema, final String filePath, final boolean reloadRefdata,
+                                    final boolean cleanseDatabase) {
+
+        try {
+
+            // database connection
+            LOGGER.info("Starting load for schema[" + targetSchema + "], filePath[" + filePath + "], reloadRefdata[" +
+                    reloadRefdata + "]");
+            final IDatabaseConnection dbConnection = getConnectionInstance(targetSchema);
+
+            // Create a DBUnit Dataset builder using the target schemas DTD
+            final FlatXmlDataSetBuilder builder = new FlatXmlDataSetBuilder();
+            builder.setDtdMetadata(true);
+            builder.setMetaDataSetFromDtd(DBUnitUtility.class.getResourceAsStream("/DTD/" + targetSchema + ".dtd"));
+            builder.setColumnSensing(true);
+
+            // Generate the clean dataset, if loadRefData is true use
+            // clean_all.xml otherwise use clean.xml
+            final String cleanDataPath = "/clean/" + targetSchema + "_clean_minus_refdata" + ".xml";
+            LOGGER.info("Reading clean dataset for data path=" + cleanDataPath);
+            final IDataSet cleanDataset = builder.build(DBUnitUtility.class.getResourceAsStream(cleanDataPath));
+
+            final IDataSet dtdDataSet =
+                    new FlatDtdDataSet(DBUnitUtility.class.getResourceAsStream("/DTD/" + targetSchema + ".dtd"));
+            final CompositeDataSet compositeDataSet = new CompositeDataSet(dtdDataSet, cleanDataset);
+
+            // Disable triggers & constraints
+            prepareForDbunitLoad(targetSchema);
```

Comment:
The other call to this method (on line 479) has been commented out.  Will this functionality be used any more?  If not this line should be removed or commented out if investigations are still ongoing.

Reply:
keepiing these for now, uncommented it now.
will delete DBUnitUtility.java, once we are happy with the changes done and having working application